code/system/application/perception/src/detectlane/detectlane.cpp

```diff
@@ -535,112 +299,181 @@ void DetectLane::GetGrouping(std::vector<cv::Vec2f> &a_groups, std::vector<cv::V
+    point1 = TransformPointToGlobalFrame(point1);
 
-    float x2 =  (t2 * a_p[i][0] + a_m[i][0] + m_roi[0]); 
+    float x2 =  (t2 * (*it).first[0] + (*it).second[0] + m_roi[0]); 
     point2 << x2, m_lowerLaneLimit, 1;
-    TransformPointToGlobalFrame(point2);
+    point2 = TransformPointToGlobalFrame(point2);
 
-    a_xWorldP.push_back(cv::Vec2f(point1(0),point2(0)));
-    a_yWorldP.push_back(cv::Vec2f(point1(1),point2(1)));
+    m_xWorldP.push_back(cv::Vec2f(point1(0),point2(0)));
+    m_yWorldP.push_back(cv::Vec2f(point1(1),point2(1)));
 
-    a_xScreenP.push_back(cv::Vec2f(x1,x2));
-    a_yScreenP.push_back(cv::Vec2f(m_upperLaneLimit,m_lowerLaneLimit));
+    m_xScreenP.push_back(cv::Vec2f(x1,x2));
+    m_yScreenP.push_back(cv::Vec2f(m_upperLaneLimit,m_lowerLaneLimit));
     
   }
 }
-void DetectLane::GetLinePairs(std::vector<cv::Vec2f> &a_xPoints
-  , std::vector<int8_t> &a_groupIds)
+
+
+
+void DetectLane::UpdateVisualLines()
 {
-  // float leftId, rightId;
-  // for(uint16_t i = 0; i < a_xPoints.size()-1; i++) {
-  //   for(uint16_t j= i+1; j < a_xPoints.size(); j++) {
-  //     if(a_xPoints[i][0] < a_xPoints[j][1]) {
-  //       leftId = i, rightId = j;
-  //     } else {
-  //       leftId = j, rightId = i;
-  //     }
-  //     a_groupIds.push_back(cv::Vec2f(leftId,rightId));
-      
-  //     // float xDiff1 = xPoints[rightId][0] - xPoints[leftId][0];
-  //     // float xDiff2 = xPoints[rightId][1] - xPoints[leftId][1];
-  //     // Lane width has to be  2 < w < 5 to be valid
-  //     //if( xDiff1 > 0 && xDiff1 < 10 && xDiff2 > 0 && xDiff2 < 10){
-  //     //}
-  //   }
+  // Canny transformation 
+  cv::Canny(m_visualMemory.back().second, m_cannyImg, m_cannyThreshold, m_cannyThreshold*3, 3);
+  cv::Sobel(m_cannyImg, m_cannyImg, m_cannyImg.depth(), 1, 0, 3, 1, 0, cv::BORDER_DEFAULT);
+
+  // Adaptive Threshold
+  cv::cvtColor(m_visualMemory.back().second, m_adapThreshImg, CV_RGB2GRAY);
+  cv::adaptiveThreshold(m_adapThreshImg,m_adapThreshImg,255,cv::ADAPTIVE_THRESH_GAUSSIAN_C, cv::THRESH_BINARY_INV, m_adapThreshKernelSize, m_adapThreshConst);
+  cv::Sobel(m_adapThreshImg, m_adapThreshImg, m_adapThreshImg.depth(), 1, 0, 3, 1, 0, cv::BORDER_DEFAULT); 
+
+  // Vector holder for each line (rho,theta)
+  cv::vector<cv::Vec2f> detectedLinesCanny;
+  cv::vector<cv::Vec2f> detectedLinesAdapThresh;
+
+  // Hough Transform 
+  // OpenCV function that uses the Hough transform and finds the "strongest" lines in the transformation    
+  cv::HoughLines(m_cannyImg, detectedLinesCanny, 1, opendlv::Constants::PI/180.0, m_houghThreshold);
+  cv::HoughLines(m_adapThreshImg, detectedLinesAdapThresh, 1, opendlv::Constants::PI/180.0, m_houghThreshold);
+  //cv::HoughLinesP(TresholdRes, detectedLinesAdapThresh, 1, opendlv::Constants::PI/180, 80, 30, 10 );
+  // std::cout << "From Hough, detectedLinesCanny: "<< detectedLinesCanny.size() << std::endl;
+  // std::cout << "From Hough, detectedLinesAdapThresh: "<< detectedLinesAdapThresh.size() << std::endl;
```

Comment:
Consider to remove unused lines (if it's not very likely that they will be included later).